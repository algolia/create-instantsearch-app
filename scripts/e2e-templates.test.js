const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');
const walkSync = require('walk-sync');
const { toMatchImageSnapshot } = require('jest-image-snapshot');

expect.extend({ toMatchImageSnapshot });

const templatesFolder = path.join(__dirname, '../src/templates');
const templates = fs
  .readdirSync(templatesFolder)
  .map(name => path.join(templatesFolder, name))
  .filter(source => fs.lstatSync(source).isDirectory());

describe('Templates', () => {
  templates.forEach(templatePath => {
    const templateName = path.basename(templatePath);
    const templateConfig = require(`${templatePath}/.template.js`);

    describe(templateName, () => {
      let temporaryDirectory;
      let appPath;
      let configFilePath;
      let generatedFiles;

      beforeAll(() => {
        temporaryDirectory = execSync(
          'mktemp -d 2>/dev/null || mktemp -d -t "appPath"'
        )
          .toString()
          .trim();

        appPath = `${temporaryDirectory}/${templateConfig.appName}`;

        const config = {
          name: `${templateConfig.appName}`,
          template: templateName,
          libraryVersion: '1.0.0',
          appId: 'appId',
          apiKey: 'apiKey',
          indexName: 'indexName',
          searchPlaceholder: 'Search placeholder',
          mainAttribute: 'mainAttribute',
          attributesForFaceting: ['facet1', 'facet2'],
        };

        configFilePath = `${temporaryDirectory}/${
          templateConfig.appName
        }.config.json`;

        fs.writeFileSync(configFilePath, JSON.stringify(config));

        execSync(
          `yarn start ${appPath} \
              --config ${configFilePath} \
              --no-installation`,
          { stdio: 'ignore' }
        );

        const ignoredFiles = fs
          .readFileSync(`${appPath}/.gitignore`)
          .toString()
          .split('\n')
          .filter(line => !line.startsWith('#'))
          .filter(Boolean)
          .concat('.DS_Store');

        generatedFiles = walkSync(appPath, {
          directories: false,
          ignore: ignoredFiles,
        });
      });

      afterAll(() => {
        execSync(`rm -rf "${temporaryDirectory}"`);
      });

      test('Folder structure', () => {
        expect(generatedFiles).toMatchSnapshot('contains the right files');
      });

      test('File content', () => {
        generatedFiles.forEach(filePath => {
          if (['.png', '.ico', '.jpg'].includes(filePath.slice(-4))) {
            const image = fs.readFileSync(`${templatePath}/${filePath}`);

            expect(image).toMatchImageSnapshot({
              customSnapshotIdentifier: `e2e-installs-${
                templateConfig.templateName
              }-${path.basename(filePath)}`,
            });
          } else {
            const fileContent = fs
              .readFileSync(`${appPath}/${filePath}`)
              .toString()
              .trim();

            expect(fileContent).toMatchSnapshot(filePath);
          }
        });
      });
    });
  });
});
