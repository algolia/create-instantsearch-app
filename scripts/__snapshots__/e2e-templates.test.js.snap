// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates InstantSearch.js File content: .gitignore 1`] = `
"# See https://help.github.com/ignore-files/ for more about ignoring files.

# dependencies
/node_modules

# testing
/coverage

# production
/dist
/.cache

# misc
.DS_Store
.env.local
.env.development.local
.env.test.local
.env.production.local

npm-debug.log*
yarn-debug.log*
yarn-error.log*"
`;

exports[`Templates InstantSearch.js File content: README.md 1`] = `
"# instantsearch.js-app

_This project was generated with [create-instantsearch-app](https://github.com/algolia/create-instantsearch-app) by [Algolia](https://algolia.com)._

## Get started

To run this project locally, install the dependencies and run the local server:

\`\`\`sh
npm install
npm start
\`\`\`

Alternatively, you may use [Yarn](https://http://yarnpkg.com/):

\`\`\`sh
yarn
yarn start
\`\`\`"
`;

exports[`Templates InstantSearch.js File content: index.html 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">

<head>
  <meta charset=\\"utf-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, shrink-to-fit=no\\">
  <meta name=\\"theme-color\\" content=\\"#000000\\">

  <link rel=\\"manifest\\" href=\\"./manifest.webmanifest\\">
  <link rel=\\"shortcut icon\\" href=\\"./favicon.png\\">

  <link rel=\\"stylesheet\\" href=\\"https://cdn.jsdelivr.net/npm/instantsearch.js@1.0.0/dist/instantsearch.min.css\\">
  <link rel=\\"stylesheet\\" href=\\"https://cdn.jsdelivr.net/npm/instantsearch.js@1.0.0/dist/instantsearch-theme-algolia.min.css\\">
  <link rel=\\"stylesheet\\" href=\\"./src/index.css\\">
  <link rel=\\"stylesheet\\" href=\\"./src/app.css\\">

  <title>instantsearch.js-app</title>
</head>

<body>
  <header class=\\"header\\">
    <h1 class=\\"header-title\\">
      <a href=\\"/\\">instantsearch.js-app</a>
    </h1>
    <p class=\\"header-subtitle\\">using InstantSearch.js</p>
  </header>

  <div class=\\"container\\">
    <div class=\\"container-app\\">
      <div class=\\"filters\\">
        <div id=\\"facet1-list\\"></div>
        <div id=\\"facet2-list\\"></div>
      </div>

      <div class=\\"search-container\\">
        <div id=\\"searchBox\\"></div>
        <div id=\\"hits\\"></div>
      </div>
    </div>

    <div id=\\"pagination\\"></div>
  </div>

  <script src=\\"https://cdn.jsdelivr.net/npm/instantsearch.js@1.0.0/dist/instantsearch.min.js\\"></script>
  <script src=\\"./src/app.js\\"></script>
</body>

</html>"
`;

exports[`Templates InstantSearch.js File content: manifest.webmanifest 1`] = `
"{
  \\"short_name\\": \\"instantsearch.js-app\\",
  \\"name\\": \\"instantsearch.js-app Sample\\",
  \\"icons\\": [
    {
      \\"src\\": \\"favicon.png\\",
      \\"sizes\\": \\"64x64 32x32 24x24 16x16\\",
      \\"type\\": \\"image/x-icon\\"
    }
  ],
  \\"start_url\\": \\"./index.html\\",
  \\"display\\": \\"standalone\\",
  \\"theme_color\\": \\"#000000\\",
  \\"background_color\\": \\"#ffffff\\"
}"
`;

exports[`Templates InstantSearch.js File content: package.json 1`] = `
"{
  \\"name\\": \\"instantsearch.js-app\\",
  \\"version\\": \\"1.0.0\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"start\\": \\"parcel index.html --port 3000\\",
    \\"build\\": \\"parcel build index.html\\"
  },
  \\"devDependencies\\": {
    \\"parcel-bundler\\": \\"^1.8.1\\"
  }
}"
`;

exports[`Templates InstantSearch.js File content: src/app.css 1`] = `
"em {
  background: cyan;
  font-style: normal;
}

.header {
  display: flex;
  align-items: center;
  min-height: 50px;
  padding: 0.5rem 1rem;
  background-image: linear-gradient(284deg, #fedd4e, #fcb43a);
  color: #fff;
  margin-bottom: 1rem;
}

.header-title {
  font-size: 1.2rem;
  font-weight: normal;
}

.header-title::after {
  content: ' ▸ ';
  padding: 0 0.5rem;
}

.header-title a {
  color: #fff;
  text-decoration: none;
}

.header-subtitle {
  font-size: 1.2rem;
}

.container {
  max-width: 1200px;
  margin: 0 auto;
  padding: 1rem;
}

.container-app {
  display: grid;
  grid-template-columns: 20% 75%;
  grid-gap: 5%;
}

.ais-hits {
  display: grid;
  grid-template-columns: 47.5% 47.5%;
  grid-gap: 1rem;
}

.ais-hits--item {
  min-height: 100px;
  padding: 1rem;
  background: #fff;
  border-radius: 4px;
  border: 1px solid rgba(150, 150, 150, 0.16);
  box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.16);
}

#searchBox {
  margin-bottom: 2rem;
}

#pagination {
  margin: 2rem auto;
  text-align: center;
}"
`;

exports[`Templates InstantSearch.js File content: src/app.js 1`] = `
"/* global instantsearch */

const search = instantsearch({
  appId: 'appId',
  apiKey: 'apiKey',
  indexName: 'indexName',
});

search.addWidget(
  instantsearch.widgets.searchBox({
    container: '#searchBox',
    placeholder: 'Search placeholder',
  })
);

search.addWidget(
  instantsearch.widgets.hits({
    container: '#hits',
    templates: {
      item: '<div class=\\"hit\\">{{{_highlightResult.mainAttribute.value}}}</div>',
    },
  })
);

search.addWidget(
  instantsearch.widgets.refinementList({
    container: '#facet1-list',
    attributeName: 'facet1',
  })
);

search.addWidget(
  instantsearch.widgets.refinementList({
    container: '#facet2-list',
    attributeName: 'facet2',
  })
);


search.addWidget(
  instantsearch.widgets.pagination({
    container: '#pagination',
  })
);

search.start();"
`;

exports[`Templates InstantSearch.js File content: src/index.css 1`] = `
"body,
h1 {
  margin: 0;
  padding: 0;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica,
    Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
}"
`;

exports[`Templates InstantSearch.js Folder structure: contains the right files 1`] = `
Array [
  ".gitignore",
  "README.md",
  "favicon.png",
  "index.html",
  "manifest.webmanifest",
  "package.json",
  "src/app.css",
  "src/app.js",
  "src/index.css",
]
`;

exports[`Templates React InstantSearch File content: .gitignore 1`] = `
"# See https://help.github.com/ignore-files/ for more about ignoring files.

# dependencies
/node_modules

# testing
/coverage

# production
/build

# misc
.DS_Store
.env.local
.env.development.local
.env.test.local
.env.production.local

npm-debug.log*
yarn-debug.log*
yarn-error.log*"
`;

exports[`Templates React InstantSearch File content: README.md 1`] = `
"# react-instantsearch-app

_This project was generated with [create-instantsearch-app](https://github.com/algolia/create-instantsearch-app) by [Algolia](https://algolia.com)._

## Get started

To run this project locally, install the dependencies and run the local server:

\`\`\`sh
npm install
npm start
\`\`\`

Alternatively, you may use [Yarn](https://http://yarnpkg.com/):

\`\`\`sh
yarn
yarn start
\`\`\`"
`;

exports[`Templates React InstantSearch File content: package.json 1`] = `
"{
  \\"name\\": \\"react-instantsearch-app\\",
  \\"version\\": \\"1.0.0\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"start\\": \\"react-scripts start\\",
    \\"build\\": \\"react-scripts build\\"
  },
  \\"dependencies\\": {
    \\"react\\": \\"^16.3.2\\",
    \\"react-dom\\": \\"^16.3.2\\",
    \\"react-instantsearch\\": \\"^1.0.0\\",
    \\"react-scripts\\": \\"1.1.4\\"
  },
  \\"devDependencies\\": {
    \\"prop-types\\": \\"^15.6.1\\"
  }
}"
`;

exports[`Templates React InstantSearch File content: public/index.html 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">

<head>
  <meta charset=\\"utf-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, shrink-to-fit=no\\">
  <meta name=\\"theme-color\\" content=\\"#000000\\">

  <link rel=\\"manifest\\" href=\\"%PUBLIC_URL%/manifest.json\\">
  <link rel=\\"shortcut icon\\" href=\\"%PUBLIC_URL%/favicon.png\\">

  <!--
    Do not use @7 in production, use a complete version like x.x.x, see website for latest version:
    https://community.algolia.com/react-instantsearch/Getting_started.html#load-the-algolia-theme
  -->
  <link rel=\\"stylesheet\\" href=\\"https://cdn.jsdelivr.net/npm/instantsearch.css@7/themes/algolia-min.css\\">

  <title>react-instantsearch-app</title>
</head>

<body>
  <noscript>
    You need to enable JavaScript to run this app.
  </noscript>

  <div id=\\"root\\"></div>
</body>

</html>"
`;

exports[`Templates React InstantSearch File content: public/manifest.json 1`] = `
"{
  \\"short_name\\": \\"react-instantsearch-app\\",
  \\"name\\": \\"react-instantsearch-app Sample\\",
  \\"icons\\": [
    {
      \\"src\\": \\"favicon.png\\",
      \\"sizes\\": \\"64x64 32x32 24x24 16x16\\",
      \\"type\\": \\"image/x-icon\\"
    }
  ],
  \\"start_url\\": \\"./index.html\\",
  \\"display\\": \\"standalone\\",
  \\"theme_color\\": \\"#000000\\",
  \\"background_color\\": \\"#ffffff\\"
}"
`;

exports[`Templates React InstantSearch File content: src/App.css 1`] = `
"em {
  background: cyan;
  font-style: normal;
}

.header {
  display: flex;
  align-items: center;
  min-height: 50px;
  padding: 0.5rem 1rem;
  background-image: linear-gradient(to right, #8e43e7, #00aeff);
  color: #fff;
  margin-bottom: 1rem;
}

.header a {
  color: #fff;
  text-decoration: none;
}

.header-title {
  font-size: 1.2rem;
  font-weight: normal;
}

.header-title::after {
  content: ' ▸ ';
  padding: 0 0.5rem;
}

.header-subtitle {
  font-size: 1.2rem;
}

.container {
  max-width: 1200px;
  margin: 0 auto;
  padding: 1rem;
}

.container-app {
  display: grid;
  grid-template-columns: 20% 75%;
  grid-gap: 5%;
}

.searchBox {
  margin-bottom: 2rem;
}

.pagination {
  margin: 2rem auto;
  text-align: center;
}"
`;

exports[`Templates React InstantSearch File content: src/App.js 1`] = `
"import React, { Component } from 'react';
import {
  InstantSearch,
  Hits,
  SearchBox,
  RefinementList,
  Pagination,
  Highlight,
} from 'react-instantsearch/dom';
import PropTypes from 'prop-types';
import './App.css';

class App extends Component {
  render() {
    return (
      <div>
        <header className=\\"header\\">
          <h1 className=\\"header-title\\">
            <a href=\\"/\\">react-instantsearch-app</a>
          </h1>
          <p className=\\"header-subtitle\\">
            using{' '}
            <a href=\\"https://github.com/algolia/react-instantsearch\\">
              React InstantSearch
            </a>
          </p>
        </header>

        <div className=\\"container\\">
          <InstantSearch
            appId=\\"appId\\"
            apiKey=\\"apiKey\\"
            indexName=\\"indexName\\"
          >
            <div className=\\"container-app\\">
              <div className=\\"filters\\">
                <RefinementList attribute=\\"facet1\\" />
                <RefinementList attribute=\\"facet2\\" />
              </div>

              <div className=\\"search-container\\">
                <SearchBox className=\\"searchBox\\" placeholder=\\"Search placeholder\\" />
                <Hits hitComponent={Hit} />

                <div className=\\"pagination\\">
                  <Pagination />
                </div>
              </div>
            </div>
          </InstantSearch>
        </div>
      </div>
    );
  }
}

function Hit(props) {
  return (
    <div>
      <Highlight attribute=\\"mainAttribute\\" hit={props.hit} />
    </div>
  );
}

Hit.propTypes = {
  hit: PropTypes.object.isRequired,
};

export default App;"
`;

exports[`Templates React InstantSearch File content: src/index.css 1`] = `
"body,
h1 {
  margin: 0;
  padding: 0;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica,
    Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
}"
`;

exports[`Templates React InstantSearch File content: src/index.js 1`] = `
"import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import App from './App';

ReactDOM.render(<App />, document.getElementById('root'));"
`;

exports[`Templates React InstantSearch Folder structure: contains the right files 1`] = `
Array [
  ".gitignore",
  "README.md",
  "package.json",
  "public/favicon.png",
  "public/index.html",
  "public/manifest.json",
  "src/App.css",
  "src/App.js",
  "src/index.css",
  "src/index.js",
]
`;

exports[`Templates Vue InstantSearch File content: .babelrc 1`] = `
"{
  \\"presets\\": [
    [\\"env\\", { \\"modules\\": false }],
    \\"stage-3\\"
  ]
}"
`;

exports[`Templates Vue InstantSearch File content: .gitignore 1`] = `
"node_modules/
npm-debug.log
yarn-debug.log
yarn-error.log
dist/"
`;

exports[`Templates Vue InstantSearch File content: README.md 1`] = `
"# vue-instantsearch-app

_This project was generated with [create-instantsearch-app](https://github.com/algolia/create-instantsearch-app) by [Algolia](https://algolia.com)._

## Get started

To run this project locally, install the dependencies and run the local server:

\`\`\`sh
npm install
npm start
\`\`\`

Alternatively, you may use [Yarn](https://http://yarnpkg.com/):

\`\`\`sh
yarn
yarn start
\`\`\`"
`;

exports[`Templates Vue InstantSearch File content: index.html 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">

<head>
  <meta charset=\\"utf-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, shrink-to-fit=no\\">
  <meta name=\\"theme-color\\" content=\\"#000000\\">

  <link rel=\\"manifest\\" href=\\"manifest.json\\">
  <link rel=\\"shortcut icon\\" href=\\"favicon.png\\">

  <!--
    Do not use @7 in production, use a complete version like x.x.x, see website for latest version:
    https://community.algolia.com/react-instantsearch/Getting_started.html#load-the-algolia-theme
  -->
  <link rel=\\"stylesheet\\" href=\\"https://cdn.jsdelivr.net/npm/instantsearch.css@7/themes/algolia-min.css\\">

  <title>vue-instantsearch-app</title>
</head>

<body>
  <noscript>
    You need to enable JavaScript to run this app.
  </noscript>

  <div id=\\"app\\"></div>

  <script src=\\"/dist/build.js\\"></script>
</body>

</html>"
`;

exports[`Templates Vue InstantSearch File content: manifest.json 1`] = `
"{
  \\"short_name\\": \\"vue-instantsearch-app\\",
  \\"name\\": \\"vue-instantsearch-app Sample\\",
  \\"icons\\": [
    {
      \\"src\\": \\"favicon.png\\",
      \\"sizes\\": \\"64x64 32x32 24x24 16x16\\",
      \\"type\\": \\"image/x-icon\\"
    }
  ],
  \\"start_url\\": \\"./index.html\\",
  \\"display\\": \\"standalone\\",
  \\"theme_color\\": \\"#000000\\",
  \\"background_color\\": \\"#ffffff\\"
}"
`;

exports[`Templates Vue InstantSearch File content: package.json 1`] = `
"{
  \\"name\\": \\"vue-instantsearch-app\\",
  \\"version\\": \\"1.0.0\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"start\\": \\"cross-env NODE_ENV=development webpack-dev-server --port 3000 --hot\\",
    \\"build\\": \\"cross-env NODE_ENV=production webpack --progress --hide-modules\\"
  },
  \\"dependencies\\": {
    \\"vue\\": \\"^2.5.16\\",
    \\"vue-instantsearch\\": \\"^1.6.0\\"
  },
  \\"devDependencies\\": {
    \\"babel-core\\": \\"6.26.0\\",
    \\"babel-loader\\": \\"7.1.4\\",
    \\"babel-preset-env\\": \\"1.6.1\\",
    \\"babel-preset-stage-3\\": \\"6.24.1\\",
    \\"cross-env\\": \\"5.1.4\\",
    \\"css-loader\\": \\"0.28.11\\",
    \\"file-loader\\": \\"1.1.11\\",
    \\"vue-loader\\": \\"14.2.2\\",
    \\"vue-template-compiler\\": \\"2.5.16\\",
    \\"webpack\\": \\"3.11.0\\",
    \\"webpack-dev-server\\": \\"2.11.2\\"
  }
}"
`;

exports[`Templates Vue InstantSearch File content: src/App.vue 1`] = `
"<template>
  <div>
    <header class=\\"header\\">
      <h1 class=\\"header-title\\">
        <a href=\\"/\\">vue-instantsearch-app</a>
      </h1>
      <p class=\\"header-subtitle\\">
        using
        <a href=\\"https://github.com/algolia/vue-instantsearch\\">
          Vue InstantSearch
        </a>
      </p>
    </header>

    <div class=\\"container\\">
      <ais-index
        app-id=\\"appId\\"
        api-key=\\"apiKey\\"
        index-name=\\"indexName\\"
      >
        <div class=\\"container-app\\">
          <div class=\\"filters\\">
            <ais-refinement-list attribute-name=\\"facet1\\"></ais-refinement-list>
            <ais-refinement-list attribute-name=\\"facet2\\"></ais-refinement-list>
          </div>

          <div class=\\"search-container\\">
            <ais-search-box
              placeholder=\\"Search placeholder\\"
              class=\\"ais-SearchBox-form\\"
              :class-names=\\"{
                'ais-search-box': 'ais-SearchBox',
                'ais-input': 'ais-SearchBox-input',
                'ais-clear': 'ais-SearchBox-reset',
                'ais-clear--disabled': 'ais-SearchBox-reset--disabled',
                'ais-search-box__submit': 'ais-SearchBox-submit',
                'ais-search-box__loading-indicator': 'ais-SearchBox-loadingIndicator',
              }\\"
            ></ais-search-box>

            <ais-results class=\\"ais-Hits-list\\">
              <template slot-scope=\\"{ result }\\">
                <div class=\\"ais-Hits-item\\">
                  <ais-highlight :result=\\"result\\" attribute-name=\\"mainAttribute\\"></ais-highlight>
                </div>
              </template>
            </ais-results>

            <div class=\\"pagination\\">
              <ais-pagination
                :class-names=\\"{
                  'ais-pagination': 'ais-Pagination-list',
                  'ais-pagination__item': 'ais-Pagination-item',
                  'ais-pagination__item--next': 'ais-Pagination-item--next',
                  'ais-pagination__item--previous': 'ais-Pagination-item--previous',
                  'ais-pagination__item--disabled': 'ais-Pagination-item--disabled',
                  'ais-pagination__item--first': 'ais-Pagination-item--first',
                  'ais-pagination__item--last': 'ais-Pagination-item--last',
                  'ais-pagination__link': 'ais-Pagination-link',
                }\\"
              ></ais-pagination>
            </div>
          </div>
        </div>
      </ais-index>
    </div>
  </div>
</template>

<style>
body,
h1 {
  margin: 0;
  padding: 0;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica,
    Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
}

em {
  background: cyan;
  font-style: normal;
}

.header {
  display: flex;
  align-items: center;
  min-height: 50px;
  padding: 0.5rem 1rem;
  background-image: linear-gradient(to right, #4dba87, #2f9088);
  color: #fff;
  margin-bottom: 1rem;
}

.header a {
  color: #fff;
  text-decoration: none;
}

.header-title {
  font-size: 1.2rem;
  font-weight: normal;
}

.header-title::after {
  content: ' ▸ ';
  padding: 0 0.5rem;
}

.header-subtitle {
  font-size: 1.2rem;
}

.container {
  max-width: 1200px;
  margin: 0 auto;
  padding: 1rem;
}

.container-app {
  display: grid;
  grid-template-columns: 20% 75%;
  grid-gap: 5%;
}

.searchBox {
  margin-bottom: 2rem;
}

.pagination {
  margin: 2rem auto;
  text-align: center;
}
</style>"
`;

exports[`Templates Vue InstantSearch File content: src/main.js 1`] = `
"import Vue from 'vue';
import App from './App.vue';
import InstantSearch from 'vue-instantsearch';

Vue.use(InstantSearch);

new Vue({
  el: '#app',
  render: h => h(App),
});"
`;

exports[`Templates Vue InstantSearch File content: webpack.config.js 1`] = `
"const path = require('path');
const webpack = require('webpack');

module.exports = {
  entry: './src/main.js',
  output: {
    path: path.resolve(__dirname, './dist'),
    publicPath: '/dist/',
    filename: 'build.js',
  },
  module: {
    rules: [
      {
        test: /\\\\.css$/,
        use: ['vue-style-loader', 'css-loader'],
      },
      {
        test: /\\\\.vue$/,
        loader: 'vue-loader',
        options: {
          loaders: {},
          // other vue-loader options go here
        },
      },
      {
        test: /\\\\.js$/,
        loader: 'babel-loader',
        exclude: /node_modules/,
      },
      {
        test: /\\\\.(png|jpg|gif|svg)$/,
        loader: 'file-loader',
        options: {
          name: '[name].[ext]?[hash]',
        },
      },
    ],
  },
  resolve: {
    alias: {
      vue$: 'vue/dist/vue.esm.js',
    },
    extensions: ['*', '.js', '.vue', '.json'],
  },
  devServer: {
    historyApiFallback: true,
    noInfo: true,
    overlay: true,
    port: 3000,
  },
  performance: {
    hints: false,
  },
  devtool: '#eval-source-map',
};

if (process.env.NODE_ENV === 'production') {
  module.exports.devtool = '#source-map';
  // http://vue-loader.vuejs.org/en/workflow/production.html
  module.exports.plugins = (module.exports.plugins || []).concat([
    new webpack.DefinePlugin({
      'process.env': {
        NODE_ENV: '\\"production\\"',
      },
    }),
    new webpack.optimize.UglifyJsPlugin({
      sourceMap: true,
      compress: {
        warnings: false,
      },
    }),
    new webpack.LoaderOptionsPlugin({
      minimize: true,
    }),
  ]);
}"
`;

exports[`Templates Vue InstantSearch Folder structure: contains the right files 1`] = `
Array [
  ".babelrc",
  ".gitignore",
  "README.md",
  "favicon.png",
  "index.html",
  "manifest.json",
  "package.json",
  "src/App.vue",
  "src/main.js",
  "webpack.config.js",
]
`;
